### ISSUE TITLE
SOQL Datatable - questions about Action Config

### ISSUE THREAD

**Issue by solo-1234** [tsalb/lwc-utils#37]
Hi! I have a number of questions about setting up Action Configurations, figured I can ask them here to get answers and maybe get the readme updated too if you think it should be. Thanks!

1. On Datatable Config, is Type supposed to contain Actions / Lookups / both? It’s not clear from the tooltip what that field is for. (Why not make it a picklist so it’s easier to understand? )

2. On Datatable Action Config, what do the different Types do? I'm guessing that Row Actions are for selecting record(s) and running something, but what are the others?

3. Do I need to choose a Row Action Name? Would that be only if the Type is Row Action?

4. For Flow API name and LWC Name – am I supposed to enter just one of those? Should it be a Flow if Row Action is custom_flow and same if LWC?

I can kind of try to guess the answers to these by looking at the example configs, but I think it would be a better experience if the tooltips and/or the readme gave more info about them.

Thank you!

**Comment by tsalb** [comment]
Good questions!

1) Picklist isn't correct, if anything, a Multi-Select Picklist is the actual functionality. In other words, `Actions;Lookups` is a valid value and so is just `Actions` and `Lookups` individually.

2) The `Table Actions` are the top right buttons (which allows you to multi-select rows). Depending on if you want those buttons to point to a Flow or LWC action, you choose and fill in the correct one just like how the recipes show you.

3) Yes, please refer to the README image (under Configurable Flow or LWC Actions) or the recipes. It's much easier to just follow the recipe which is why I made them!

4) Please refer to the recipe - it shows various combinations. There are some validation rules in place but I probably didn't catch them all.

My intention was always to have users install the recipes to play with it, the tooltip/readme would make it much more complicated to understand (simply by virtue of the fact there are too many permutations, and walls of text are hard to understand).

So then, picture + recipes!

**Comment by solo-1234** [comment]
Thanks for answering! 

1. With just three real choices, I'd say you could have a picklist to choose Actions, Lookups, or Actions; Lookups. But that's up to you.  I'd recommend adding to the tooltip though.

2. Aha, I was thinking the difference was where the datatable was, not what type of action we're talking about. 
What's the difference between Primary and Secondary actions?

3 and 4: can I make some recommendations for tooltips, just to make things simpler? It's hard to jump back and forth between the examples and the actions I'm building, and I think it would be a better UX if we explained things right there.

**Comment by solo-1234** [comment]
Here is my suggested tooltip for the Type field:

```
Table Actions (which can call either Flow or LWC) are placed above the datatable. Choose Primary or Secondary to assign this action to one of the two slots.
Row Actions are placed on each row and accessed via the down arrow at the right.
```
(it's confusing because this field controls both where the button goes and what it does, maybe it would be simpler if "what it does" was just controlled by the fields on the right)

**Comment by tsalb** [comment]
Going to word-smith that a bit:

```
Primary / Secondary Table Actions are single/multi row actions placed above the datatable. Currently, there are only two slots.

Row Actions are placed on each row and accessed via the action menu dropdown on the far right for each row.
```

**Comment by solo-1234** [comment]
That works, but doesn't explain when you'd choose Flow vs LWC

**Comment by solo-1234** [comment]
If you are planning for a 2.0 soonish then it might pay to just wait, and then you can change Type to Display Type and have Table Action - Primary, Table Action - Secondary, and Row Action

**Comment by tsalb** [comment]
I think that is self explanatory when you open the picklist of options for `Action_Config.Type__c`, and coupled with a validation rule (which i think is already in place) should be pretty straightforward.

That's the rationale for omitting the `(which can call either Flow or LWC)` piece in the help text.

As far as 2.0, no plans at all (definitely not in the next 2 quarters)

**Comment by solo-1234** [comment]
... and I thought that the flow/lwc part was referring to where the datatable was displayed - in a flow/page

**Comment by solo-1234** [comment]
Suggested tooltip for: Row Action Name
`If you choose custom_lwc or custom_flow, enter the relevant name below.`

Suggested tooltip for Order: 
`This will determine the order for Row Action buttons`

**Comment by tsalb** [comment]
Ah, regarding your earlier comment I didn't even think that was a possible train of thought lol.

That's fair, in that case we can do this:

```
Primary / Secondary Table Actions, which call either Flow or LWC, are single / multi row actions placed above the datatable. Currently, there are only two slots.

Row Actions are accessed via the action menu dropdown on the far right for each individual row.
```

**Comment by solo-1234** [comment]
Nice! And you can cut out that line when you remove the flow/lwc choice from that field in 2.0

**Comment by solo-1234** [comment]
Re: Order, if you want, you can add `The order for Table Actions is determined in the Type field`. But that may be too wordy for you.

**Comment by solo-1234** [comment]
And I'd add to the Type tooltip on the parent:
`Semicolon delimited to keep track of which child configs (Lookups and/or Actions) are used for this grouping record.`

(Also wondering if you need this field or if it can just figure it out based on which child records exist.)

**Comment by tsalb** [comment]
Fair point with regards to deprecating Type, that might be a `2.0.0` thing

**Comment by solo-1234** [comment]
Makes sense
